notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-989_analysis_inertia_compensation.ipynb

```diff
@@ -0,0 +1,741 @@
+    "    )\n",
+    "     \n",
+    "    return stats"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df39fa21-262a-4446-8520-977d8b2441be",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "#### get_tma_slew_events"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "874a4aac-e211-486f-b824-e038c2195ba1",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def get_tma_slew_events(dataset : SlewDataSet, client : EfdClient | None = None, t_pad : int = 1):\n",
+    "    \"\"\"\n",
+    "    Retrieve Telescope Mount Assembly (TMA) slew events within a specified time range.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    dataset : SlewDataSet\n",
+    "        Dataset containing the begin and the end of a slew. \n",
+    "    client : EfdClient | None, optional\n",
+    "        An instance of EfdClient for connecting to the Event Forwarding Database. Default is None.\n",
+    "    t_pad : int, default=1\n",
+    "        Time padding in seconds to be added at the beginning and at the end of a slew event.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    list\n",
+    "        A list of TMA slew events that occurred within the specified time range.\n",
+    "\n",
+    "    Notes\n",
+    "    -----\n",
+    "    This function retrieves TMA slew events occurring between the specified start and end times.\n",
+    "    It uses the TMAEventMaker class to obtain events for the specified day of observation (dayObs).\n",
+    "    The events are filtered to include only those that start after 1 second before the specified start time\n",
+    "    and end before 1 second after the specified end time.\n",
+    "\n",
+    "    Example\n",
+    "    -------\n",
+    "    >>> start_time = \"2023-08-09T00:00:00\"\n",
+    "    >>> end_time = \"2023-08-09T23:59:59\"\n",
+    "    >>> tma_events = get_tma_slew_events(start_time, end_time)\n",
+    "    \"\"\"\n",
+    "    dayObs = getDayObsForTime(dataset.begin)\n",
+    "    print(dataset.begin, dayObs)\n",
+    "    \n",
+    "    eventMaker = TMAEventMaker()\n",
+    "    events = eventMaker.getEvents(dayObs)\n",
+    "    \n",
+    "    events = [e for e in events if (e.begin > dataset.begin - TimeDelta(t_pad, format=\"sec\"))\n",
+    "              & ((e.end < dataset.end + TimeDelta(t_pad, format=\"sec\")))]\n",
+    "    \n",
+    "    return events"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49cde41d-a015-482d-90be-a62368b45d2f",
+   "metadata": {},
+   "source": [
+    "## Data Collection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c510e63-2ef5-4480-a7ff-990749fee22d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Dataset 1 - Az Slew from -100 to 100 with 80 El, 30% motion settings and Inertia Compensation On\n",
+    "begin = \"2023-08-02T22:02:30\"\n",
```

Comment:
This can be improved. You are passing multiple times the same parameters - start, end times, and label which includes % of max motion. Then some stat is calculated, and added to the class, to be printed later. Can that be made to a function,taking three parameters - start. end, and label - returning datataset class? Also pass Time once, and use it anytime as Time - no need for conversion then. Don't store time in str except for passing user-supplied arguments.

Reply:
FIxed by moving all the functions to methods of the class.